src/robot/running/bodyrunner.py

```diff
@@ -556,3 +565,81 @@ def _run_finally(self, data, run):
                 return err
             else:
                 return None
+
+
+class WhileLimitExceeded(RuntimeError):
+    pass
+
+
+class WhileLimit:
+
+    @staticmethod
+    def parse_iteration_count(limit):
+        try:
+            limit = int(limit)
+            if limit <= 0:
+                raise ValueError
+            return limit
+        except ValueError:
+            raise ValueError("Iteration limit must be positive integer when using "
+                             f"'x' or 'times', got: '{limit}'")
+
+    @classmethod
+    def create(cls, limit, variables):
+        try:
+            if not limit:
+                return IterationCountLimit(100)
+            value = limit.replace('limit=', '')
```

Comment:
Why does the value contain `limit=` prefix here? Shouldn't only the value be returned by the parsing model? I'll add a separate comment there.